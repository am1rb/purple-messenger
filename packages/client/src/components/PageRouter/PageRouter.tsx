import React from "react";
import { Route, Switch } from "react-router-dom";
import withAuthPage from "features/auth/components/withAuthPage";
import withGuestPage from "features/auth/components/withGuestPage";
import routes from "core/routes";
import { RouteAuthMode } from "core/type/route";

const routeAuthModeMap = {
  [RouteAuthMode.Auth]: withAuthPage,
  [RouteAuthMode.Guest]: withGuestPage,
  [RouteAuthMode.None]: null,
};

const authRoutes = routes.map((route) => {
  const authFunc = routeAuthModeMap[route.authMode];
  const item = {
    ...route,
    component: authFunc ? authFunc!(route.component! as any) : route.component,
  };
  return item;
});

function PageRouter() {
  return (
    <Switch>
      {authRoutes.map((route, idx) => (
        <Route key={idx} {...route} />
      ))}
    </Switch>
  );
}

export default PageRouter;
