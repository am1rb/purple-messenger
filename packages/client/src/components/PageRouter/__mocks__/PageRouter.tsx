import React from "react";

function PageRouter() {
  return <div data-testid="mock-page-router" />;
}

export default PageRouter;
