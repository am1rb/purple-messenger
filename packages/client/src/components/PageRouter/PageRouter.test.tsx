import React from "react";
import { render } from "@testing-library/react";
import PageRouter from "./PageRouter";

describe("PageRouter Component Tests", () => {
  it("Should match the snapshot", () => {
    const { container } = render(<PageRouter />);
    expect(container).toMatchSnapshot();
  });
});
