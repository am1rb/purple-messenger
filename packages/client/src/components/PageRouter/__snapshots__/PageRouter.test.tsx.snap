// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageRouter Component Tests Should match the snapshot 1`] = `
<div>
  <div
    data-testid="mock-switch"
  >
    <div
      data-path="/conversation/@:username_/"
      data-testid="mock-route"
    />
  </div>
</div>
`;
